spring-cloud-dataflow-docs/src/main/asciidoc/streams.adoc

```diff
@@ -208,24 +208,145 @@ $ tail -f /var/folders/wn/8jxm_tbd1vj28c8vj37n900m0000gn/T/spring-cloud-dataflow
+
+
+==== Passing application properties when deploying stream
+
+The application properties can also be specified when deploying the stream. When specified during deployment, these application properties can either be specified as
+ `short-hand` property names (applicable for white-listed properties) or fully qualified property names. The application properties should have the prefix "app.<app name/label>".
+
+For example, the stream
+
+[source,bash]
+----
+dataflow:> stream create --definition "time | log" --name ticktock
+----
+
+can be deployed with application properties using the`short-hand` property names:
+
+[source,bash]
+----
+dataflow:>stream deploy ticktock --properties "app.time.fixed-delay=5,app.log.level=ERROR"
+----
+
+When using the app label,
+
+[source,bash]
+----
+stream create ticktock --definition "a: time | b: log"
+----
+
+the application properties can be defined as:
+
+[source,bash]
+----
+stream deploy ticktock --properties "app.a.fixed-delay=4,app.b.log.level=ERROR"
+----
+
+
+==== Overriding application properties during stream deployment
+
+Application properties that are defined during deployment overrides the same properties defined during the stream creation.
+
+For example, the following stream has application properties defined during stream creation:
+
+[source,bash]
+----
+dataflow:> stream create --definition "time --fixed-delay=5 | log --level=WARN" --name ticktock
+----
+
+To override these application properties, one can specify the new property values during deployment:
+
+[source,bash]
+----
+dataflow:>stream deploy ticktock --properties "app.time.fixed-delay=4,app.log.level=ERROR"
+----
+
+=== Deployment properties
+
+When deploying the stream, properties that control the deployment of the apps into the target platform are known as `deployment` properties.
+For instance, one can specify how many number of instances need to be deployed for the specific application defined in the stream using the deployment property called `count`.
+The stream partitioning properties are also passed as `deployment` properties during stream deployment. These properties are optionally provided as a comma delimted list of key=value pairs.
 
 ==== Passing instance count as deployment property
 
 If you would like to have multiple instances of an application in the stream, you
 can include a property with the deploy command:
 
-[source,bash,subs=attributes]
+[source,bash]
 ----
 dataflow:> stream deploy --name ticktock --properties "app.time.count=3"
 ----
 
+Note that `count` is the *reserved* property name by the underlying deployer. Hence, if the application also has a custom property named `count`, it is *not* supported
+ when specified during stream `deployment` as it conflicts with the `instance` count deployer property. Instead, the `count` as a custom application property can be
+ specified at the stream `creation` when it will be considered as an app property.
```

Comment:
OR it can be provided in fully-qualified form at deployment time, e.g. `app.foo.foo.count